src/components/forms/fields/CheckboxField/CheckboxField.tsx

```diff
@@ -0,0 +1,119 @@
+
+export type CheckboxFieldTitleProps = React.PropsWithChildren<{
+  className?: ClassNameArgument;
+
+  /** Whether to display the optional observation on title. */
+  titleOptional?: undefined | boolean,
+
+  /** An optional tooltip to be displayed on an info icon next to the title. */
+  titleTooltip?: undefined | string,
+}>;
+
+export const CheckboxFieldTitle = ({ className, children, titleOptional, titleTooltip }: CheckboxFieldTitleProps) => (
+  <h1 className={cx(
+    'bk',
+    cl['bk-checkbox-field__title'],
+    className,
+  )}>
+    {children}
+    {titleTooltip && (
+      <TooltipProvider tooltip={titleTooltip}>
+        <Icon icon="info" className={cl['bk-checkbox-field__title__icon']}/>
+      </TooltipProvider>
+    )}
+    {titleOptional && (
+      <small className={cl['bk-checkbox-field__title__optional']}>(Optional)</small>
+    )}
+  </h1>
+);
+
+export type CheckboxFieldProps = ComponentProps<'div'> & {
+  /** Whether this component should be unstyled. */
+  unstyled?: undefined | boolean,
+
+  /** A label to be displayed after the checkbox. */
+  label: string,
+
+  /** An optional supporting copy to be displayed under the label. */
+  sublabel?: undefined | string,
+
+  /** An optional title. */
+  title?: undefined | string,
+
+  /** An optional tooltip to be displayed on an info icon next to the title. */
+  titleTooltip?: undefined | string,
+
+  /** Whether to display the optional observation on title. */
+  titleOptional?: undefined | boolean,
+
+  /** Whether the checkbox is checked by default. Passed down to Checkbox component. */
+  defaultChecked?: undefined | boolean,
+
+  /** Whether the checkbox is checked. Passed down to Checkbox component. */
+  checked?: undefined | boolean,
+
+  /** Whether the checkbox is disabled. Passed down to Checkbox component. */
+  disabled?: undefined | boolean,
+};
+
+/**
+ * A full-fledged Checkbox field, with optional label, title, icon etc.
+ */
+export const CheckboxField = (props: CheckboxFieldProps) => {
+  const {
+    unstyled = false,
+    label = '',
+    sublabel,
+    title,
+    titleOptional,
+    titleTooltip,
+    className,
+    ...propsRest
+  } = props;
+
+  return (
+    <div className={cx(
+      'bk',
+      { [cl['bk-checkbox-field']]: !unstyled },
+      className,
+    )}>
+      {title && (
+        <CheckboxFieldTitle
+          titleOptional={titleOptional}
+          titleTooltip={titleTooltip}
+        >
+          {title}
+        </CheckboxFieldTitle>
+      )}
+      {/* biome ignore lint/a11y/noLabelWithoutControl: the `<Checkbox>` will resolve to an `<input>` */}
+      <label>
+        <Checkbox
+          checked={props.checked}
+          defaultChecked={props.defaultChecked}
+          disabled={props.disabled}
+        />
+        <span className={cl['bk-checkbox-field__label']}>
+          {label}
+          {sublabel && (
+            <div className={cl['bk-checkbox-field__label__sublabel']}>{sublabel}</div>
```

Comment:
`div` element inside of `span` can be avoided? (https://www.mrc-productivity.com/techblog/?ht_kb=css-tutorial-4-div-and-span-tags#:~:text=This%20code%20would%20appear%20on,inside%20of%20an%20inline%20element)

Reply:
Sorry again, I should have noticed earlier, but `div` element inside of `label` element also should be avoided (https://stackoverflow.com/questions/18609554/is-div-inside-label-block-correct).
Is it possible to have all `span` element inside of `label`?